### ISSUE TITLE
%100 Accuracy and 0.000 Loss in the training process

### ISSUE THREAD

**Issue by firaterdem** [open-mmlab/mmdetection#7605]
Thanks for your error report and we appreciate it a lot.

**Checklist**

1. I have searched related issues but cannot get the expected help.
2. I have read the [FAQ documentation](https://mmdetection.readthedocs.io/en/latest/faq.html) but cannot get the expected help.
3. The bug has not been fixed in the latest version.

**Describe the bug**
I have getting 100% accuracy and 0.0000 loss values from the second epoch of the training process.

**Reproduction**

1. What command or script did you run?

```none
python tools\train.py configs\balloon\config_file.py
```

2. Did you make any modifications on the code or config? Did you understand what you have modified?
I have modified the config file for custom dataset as follows:

```none
# The new config inherits a base config to highlight the necessary modification
_base_ = 'mask_rcnn_r50_caffe_fpn_mstrain-poly_1x_coco.py'

# We also need to change the num_classes in head to match the dataset's annotation
model = dict(
    roi_head=dict(
        bbox_head=dict(num_classes=1),
        mask_head=dict(num_classes=1)))

# Modify dataset related settings
dataset_type = 'CocoDataset'
classes = ('tree',)
data = dict(
    train=dict(
        img_prefix='configs/balloon/train_im/',
        classes=classes,
        ann_file='configs/balloon/anno_train.json'),
    val=dict(
        img_prefix='configs/balloon/val_im/',
        classes=classes,
        ann_file='configs/balloon/anno_val.json'),
    test=dict(
        img_prefix='configs/balloon/test_im/',
        classes=classes,
        ann_file='configs/balloon/anno_test.json'))
```

4. What dataset did you use?

I have used a custom dataset containing one class.

**Environment**

1. Please run `python mmdet/utils/collect_env.py` to collect necessary environment information and paste it here.
'tail' is not recognized as an internal or external command,
operable program or batch file.
'gcc' is not recognized as an internal or external command,
operable program or batch file.
sys.platform: win32
Python: 3.7.11 (default, Jul 27 2021, 09:42:29) [MSC v.1916 64 bit (AMD64)]
CUDA available: True
GPU 0: NVIDIA GeForce GTX 1060
CUDA_HOME: C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v10.1
NVCC: Not Available
GCC: n/a
PyTorch: 1.6.0
PyTorch compiling details: PyTorch built with:
  - C++ Version: 199711
  - MSVC 192628806
  - Intel(R) Math Kernel Library Version 2020.0.0 Product Build 20191125 for Intel(R) 64 architecture applications
  - Intel(R) MKL-DNN v1.5.0 (Git Hash e2ac1fac44c5078ca927cb9b90e1b3066a0b2ed0)
  - OpenMP 2019
  - CPU capability usage: AVX2
  - CUDA Runtime 10.1
  - NVCC architecture flags: -gencode;arch=compute_37,code=sm_37;-gencode;arch=compute_50,code=sm_50;-gencode;arch=compute_60,code=sm_60;-gencode;arch=compute_61,code=sm_61;-gencode;arch=compute_70,code=sm_70;-gencode;arch=compute_75,code=sm_75;-gencode;arch=compute_37,code=compute_37
  - CuDNN 7.6.4
  - Magma 2.5.2
  - Build settings: BLAS=MKL, BUILD_TYPE=Release, CXX_FLAGS=/DWIN32 /D_WINDOWS /GR /EHsc /w /bigobj -openmp:experimental -DNDEBUG -DUSE_FBGEMM -DUSE_VULKAN_WRAPPER, PERF_WITH_AVX=1, PERF_WITH_AVX2=1, PERF_WITH_AVX512=1, USE_CUDA=ON, USE_EXCEPTION_PTR=1, USE_GFLAGS=OFF, USE_GLOG=OFF, USE_MKL=ON, USE_MKLDNN=ON, USE_MPI=OFF, USE_NCCL=OFF, USE_NNPACK=OFF, USE_OPENMP=ON, USE_STATIC_DISPATCH=OFF,

TorchVision: 0.7.0
OpenCV: 4.5.5
MMCV: 1.4.4
MMCV Compiler: MSVC 192930137
MMCV CUDA Compiler: 10.1
MMDetection: 2.23.0+unknown

3. You may add addition that may be helpful for locating the problem, such as
    - How you installed PyTorch [e.g., pip, conda, source]
I have installed it with pip.

    - Other environment variables that may be related (such as `$PATH`, `$LD_LIBRARY_PATH`, `$PYTHONPATH`, etc.)
PATH=C:\Users\firat\anaconda3\envs\open-mmlab;C:\Users\firat\anaconda3\envs\open-mmlab\Library\mingw-w64\bin;C:\Users\firat\anaconda3\envs\open-mmlab\Library\usr\bin;C:\Users\firat\anaconda3\envs\open-mmlab\Library\bin;C:\Users\firat\anaconda3\envs\open-mmlab\Scripts;C:\Users\firat\anaconda3\envs\open-mmlab\bin;C:\Users\firat\anaconda3\condabin;C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v10.1\bin;C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v10.1\libnvvp;.;C:\Program Files (x86)\Common Files\Oracle\Java\javapath;C:\Windows\system32;C:\Windows;C:\Windows\System32\Wbem;C:\Windows\System32\WindowsPowerShell\v1.0;C:\Windows\System32\OpenSSH;C:\Program Files (x86)\NVIDIA Corporation\PhysX\Common;C:\Program Files\NVIDIA Corporation\NVIDIA NvDLISR;C:\Program Files\MATLAB\R2020a\runtime\win64;C:\Program Files\MATLAB\R2020a\bin;C:\Program Files (x86)\Windows Kits\8.1\Windows Performance Toolkit;C:\Program Files\CODA\bin;C:\Program Files\HARP\bin;C:\Program Files (x86)\Calibre2;C:\Program Files\SafeNet\Authentication\SAC\x64;C:\Program Files\SafeNet\Authentication\SAC\x32;C:\Program Files\NVIDIA Corporation\Nsight Compute 2019.4.0;C:\Program Files\Hexagon\ERDAS IMAGINE 2015;C:\Program Files\PuTTY;C:\Users\firat\AppData\Local\Programs\Python\Python37\Scripts;C:\Users\firat\AppData\Local\Programs\Python\Python37;C:\Users\firat\anaconda3;C:\Users\firat\anaconda3\Library\mingw-w64\bin;C:\Users\firat\anaconda3\Library\usr\bin;C:\Users\firat\anaconda3\Library\bin;C:\Users\firat\anaconda3\Scripts;C:\Users\firat\AppData\Local\Microsoft\WindowsApps;.;C:\Program Files\snap\bin

**Error traceback**
If applicable, paste the error trackback here.

```none
2022-04-01 14:21:09,682 - mmdet - INFO - Epoch [1][50/368]      lr: 1.978e-03, eta: 1:01:29, time: 0.845, data_time: 0.263, memory: 2392, loss_rpn_cls: 0.2092, loss_rpn_bbox: 0.0000, loss_cls: 0.0273, acc: 99.8730, loss_bbox: 0.0000, loss_mask: 0.0000, loss: 0.2365
2022-04-01 14:21:37,343 - mmdet - INFO - Epoch [1][100/368]     lr: 3.976e-03, eta: 0:50:16, time: 0.553, data_time: 0.008, memory: 2461, loss_rpn_cls: 0.0000, loss_rpn_bbox: 0.0000, loss_cls: 0.0000, acc: 100.0000, loss_bbox: 0.0000, loss_mask: 0.0000, loss: 0.0000
2022-04-01 14:22:06,017 - mmdet - INFO - Epoch [1][150/368]     lr: 5.974e-03, eta: 0:46:41, time: 0.573, data_time: 0.009, memory: 2461, loss_rpn_cls: 0.0000, loss_rpn_bbox: 0.0000, loss_cls: 0.0000, acc: 100.0000, loss_bbox: 0.0000, loss_mask: 0.0000, loss: 0.0000
2022-04-01 14:22:33,720 - mmdet - INFO - Epoch [1][200/368]     lr: 7.972e-03, eta: 0:44:20, time: 0.554, data_time: 0.010, memory: 2461, loss_rpn_cls: 0.0000, loss_rpn_bbox: 0.0000, loss_cls: 0.0000, acc: 100.0000, loss_bbox: 0.0000, loss_mask: 0.0000, loss: 0.0000
2022-04-01 14:23:01,376 - mmdet - INFO - Epoch [1][250/368]     lr: 9.970e-03, eta: 0:42:44, time: 0.553, data_time: 0.010, memory: 2461, loss_rpn_cls: 0.0000, loss_rpn_bbox: 0.0000, loss_cls: 0.0000, acc: 100.0000, loss_bbox: 0.0000, loss_mask: 0.0000, loss: 0.0000
2022-04-01 14:23:29,085 - mmdet - INFO - Epoch [1][300/368]     lr: 1.197e-02, eta: 0:41:32, time: 0.555, data_time: 0.010, memory: 2461, loss_rpn_cls: 0.0000, loss_rpn_bbox: 0.0000, loss_cls: 0.0000, acc: 100.0000, loss_bbox: 0.0000, loss_mask: 0.0000, loss: 0.0000
2022-04-01 14:23:56,981 - mmdet - INFO - Epoch [1][350/368]     lr: 1.397e-02, eta: 0:40:34, time: 0.558, data_time: 0.009, memory: 2461, loss_rpn_cls: 0.0000, loss_rpn_bbox: 0.0000, loss_cls: 0.0000, acc: 100.0000, loss_bbox: 0.0000, loss_mask: 0.0000, loss: 0.0000
2022-04-01 14:24:07,140 - mmdet - INFO - Saving checkpoint at 1 epochs
[>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>] 315/315, 5.6 task/s, elapsed: 56s, ETA:     0s2022-04-01 14:25:05,868 - mmdet - INFO - Evaluating bbox...
Loading and preparing results...
2022-04-01 14:25:05,869 - mmdet - ERROR - The testing results of the whole dataset is empty.
2022-04-01 14:25:05,870 - mmdet - INFO - Exp name: config_file.py
2022-04-01 14:25:05,870 - mmdet - INFO - Epoch(val) [1][315]
2022-04-01 14:25:47,672 - mmdet - INFO - Epoch [2][50/368]      lr: 1.668e-02, eta: 0:40:01, time: 0.830, data_time: 0.269, memory: 2532, loss_rpn_cls: 0.0000, loss_rpn_bbox: 0.0000, loss_cls: 0.0000, acc: 100.0000, loss_bbox: 0.0000, loss_mask: 0.0000, loss: 0.0000
2022-04-01 14:26:15,492 - mmdet - INFO - Epoch [2][100/368]     lr: 1.868e-02, eta: 0:39:12, time: 0.556, data_time: 0.008, memory: 2532, loss_rpn_cls: 0.0000, loss_rpn_bbox: 0.0000, loss_cls: 0.0000, acc: 100.0000, loss_bbox: 0.0000, loss_mask: 0.0000, loss: 0.0000
2022-04-01 14:26:44,186 - mmdet - INFO - Epoch [2][150/368]     lr: 2.000e-02, eta: 0:38:33, time: 0.573, data_time: 0.009, memory: 2532, loss_rpn_cls: 0.0000, loss_rpn_bbox: 0.0000, loss_cls: 0.0000, acc: 100.0000, loss_bbox: 0.0000, loss_mask: 0.0000, loss: 0.0000
2022-04-01 14:27:12,009 - mmdet - INFO - Epoch [2][200/368]     lr: 2.000e-02, eta: 0:37:51, time: 0.557, data_time: 0.010, memory: 2532, loss_rpn_cls: 0.0000, loss_rpn_bbox: 0.0000, loss_cls: 0.0000, acc: 100.0000, loss_bbox: 0.0000, loss_mask: 0.0000, loss: 0.0000
2022-04-01 14:27:40,135 - mmdet - INFO - Epoch [2][250/368]     lr: 2.000e-02, eta: 0:37:13, time: 0.562, data_time: 0.010, memory: 2532, loss_rpn_cls: 0.0000, loss_rpn_bbox: 0.0000, loss_cls: 0.0000, acc: 100.0000, loss_bbox: 0.0000, loss_mask: 0.0000, loss: 0.0000
2022-04-01 14:28:08,053 - mmdet - INFO - Epoch [2][300/368]     lr: 2.000e-02, eta: 0:36:36, time: 0.559, data_time: 0.010, memory: 2532, loss_rpn_cls: 0.0000, loss_rpn_bbox: 0.0000, loss_cls: 0.0000, acc: 100.0000, loss_bbox: 0.0000, loss_mask: 0.0000, loss: 0.0000
2022-04-01 14:28:35,959 - mmdet - INFO - Epoch [2][350/368]     lr: 2.000e-02, eta: 0:35:59, time: 0.558, data_time: 0.009, memory: 2532, loss_rpn_cls: 0.0000, loss_rpn_bbox: 0.0000, loss_cls: 0.0000, acc: 100.0000, loss_bbox: 0.0000, loss_mask: 0.0000, loss: 0.0000
```

**Bug fix**
If you have already identified the reason, you can provide the information here. If you are willing to create a PR to fix it, please also leave a comment here and that would be much appreciated!

**Comment by BIGWangYuDong** [comment]
Maybe because your dataset is too small, and the network is learned everything.

**Comment by hhaAndroid** [comment]
@firaterdem 0.02 is the value of 8 cards x 2bs, please scale your LR parameters accordingly.

**Comment by firaterdem** [comment]
@hhaAndroid Thank you for your answer. I have added a line to the config file for reducing the learning rate as follows:

`optimizer = dict(type='SGD', lr=0.00003, weight_decay=0.0001)
`

Although the losses have been reduced, the accuracy has been %100. How can I balance LR as you mentioned?

Also, I got another error after the first epoch. Do you have an opinion on this subject too? 

![Capture](https://user-images.githubusercontent.com/19222333/161495613-5b57c5de-9176-4a0c-9e44-e2e1b3c95e4e.PNG)

Thank you for your help again.

**Comment by yirutsai** [comment]
I met the same problem as well. I use one class object detection.

**Comment by echosum** [comment]
Hi. I meet a same problem. Do you fix it now?

**Comment by BIGWangYuDong** [comment]
May your dataset is too small, and the detector is overfitted. Feel free to create a new issue if you still meet problems

**Comment by Harhammer** [comment]
Probably way to late but maybe it helps someone else.
I had the same problem and I had to change **poly2mask** to **True** in the train and test pipeline

**Comment by mahilaMoghadami** [comment]
hello,
I didnt undestan what you mean? what should I do ?
I have same problem

**Comment by Harhammer** [comment]
@mahilaMoghadami open the config file, search for poly2mask and set it to true.
